docs/adrs/006-web-apis.md

```diff
@@ -0,0 +1,171 @@
+   * - `false` indicates connection is inactive.
+   */
+  readonly isConnected: () => boolean;
+
+  /** Synchronously return present injection state */
+  readonly state: () => PenumbraState;
+
+  /** Standard EventTarget methods emitting PenumbraStateEvent upon state changes */
+  readonly addEventListener: PenumbraEventTarget['addEventListener'];
+  /** Unsubscribes from the state change events. Provide the same callback as in `addEventListener` */
+  readonly removeEventListener: PenumbraEventTarget['removeEvenListener'];
+}
+```
+
+## Client API
+
+A globally shared `client` instance should be able to establish the connection with
+injected wallet providers and use the connection to query the blockchain. The API in this case would be:
+
+```ts
+import type { PenumbraService } from '@penumbra-zone/protobuf';
+import type { PromiseClient } from '@connectrpc/connect';
+
+interface PenumbraClient {
+  /**
+   * Connects to the injected provider, asks for user approval if wasn't connected before.
+   * If `providerUrl` argument is not provided, tries to connect to the first injected provider.
+   * Returns the manifest URL of the connected provider or throws an error otherwise.
+   */
+  readonly connect: (providerUrl?: string) => Promise<string>;
+
+  /**
+   * Tries to reconnect to the injected provider (the first one with `connected` state)
+   * without asking for user approval
+   */
+  readonly reconnect: (providerUrl?: string) => Promise<string>;
+
+  /** Reexports the `disconnect` function from injected provider */
+  readonly disconnect: () => Promise<void>;
+  /** Reexports the `isConnected` function from injected provider  */
+  readonly isConnected: () => boolean | undefined;
+  /** Reexports the `state` function from injected provider */
+  readonly getState: () => PenumbraState;
+  /** Provides a simplified callback interface to `PenumbraStateEvent`s. */
+  readonly onConnectionChange: (
+    cb: (connection: { origin: string; connected: boolean; state: PenumbraState }) => void,
+  ) => void;
+
+  /** Returns a new or re-used `PromiseClient<T>` for a specific `PenumbraService` */
+  readonly service: <T extends PenumbraService>(
+    service: T,
+    options: Omit<ChannelTransportOptions, 'getPort'>,
+  ) => PromiseClient<T>;
+}
+```
+
+Moreover, the `client` package might export but not limited to the following useful functions and types:
+
+```ts
+export type PenumbraManifest = Partial<chrome.runtime.ManifestV3> &
+  Required<Pick<chrome.runtime.ManifestV3, 'name' | 'version' | 'description' | 'icons'>>;
+
+export type getInjectedProvider = (penumbraOrigin: string) => Promise<PenumbraProvider>;
+
+export type getAllInjectedProviders = () => string[];
+
+export type getPenumbraManifest = (
+  penumbraOrigin: string,
+  signal?: AbortSignal,
+) => Promise<PenumbraManifest>;
+
+export type getAllPenumbraManifests = () => Record<
+  keyof (typeof window)[typeof PenumbraSymbol],
+  Promise<PenumbraManifest>
+>;
```

Comment:
It seemed to be required by previous discussion that a `PenumbraClient` object should collect methods a developer would be interested in for selecting and then initiating a provider connection

is there a reason these are broken out again?

Reply:
moved useful utility methods into the `client` with `get` prefixes